assets/app/scripts/controllers/attachPersistentVolumeClaim.js

```diff
@@ -0,0 +1,280 @@
+            title: "Deployments",
+            link: "project/" + $routeParams.project + "/browse/deployments"
+          });
+
+          var deploymentConfig = $scope.attach.deploymentConfig;
+          if (deploymentConfig) {
+            $scope.breadcrumbs.splice(2, 0, {
+              title: deploymentConfig.metadata.name,
+              link: "project/" + $routeParams.project + "/browse/deployments/" + deploymentConfig.metadata.name
+            });
+          }
+
+          var deployment = $scope.attach.deployment;
+          if (deployment) {
+            var deploymentVersion = $filter("annotation")(deployment, "deploymentVersion");
+            $scope.breadcrumbs.splice(2, 0, {
+              title: deploymentVersion ? "#" + deploymentVersion : deployment.metadata.name
+            });
+            var deploymentDeploymentConfigName = $filter("annotation")(deployment, "deploymentConfig");
+            if (deploymentDeploymentConfigName) {
+              $scope.breadcrumbs[2].link = "project/" + $routeParams.project + "/browse/deployments/" + deploymentDeploymentConfigName + "/" + deployment.metadata.name;
+              $scope.breadcrumbs.splice(2, 0, {
+                title: deploymentDeploymentConfigName,
+                link: "project/" + $routeParams.project + "/browse/deployments/" + deploymentDeploymentConfigName
+              });
+            }
+          }
+        };
+
+        var orderByDisplayName = $filter('orderByDisplayName');
+        var getErrorDetails = $filter('getErrorDetails');
+
+        if (!validate() || !load()) {
+          return;
+        }
+
+        $scope.attachPersistentVolumeClaim = function() {
+          $scope.disableInputs = true;
+
+          if ($scope.attachPersistentVolumeClaimForm.$valid) {
+            // generate a volume name if not provided
+            if (!$scope.attach.volumeName) {
+              $scope.attach.volumeName = StorageService.generateSimpleName("volume-");
+            }
+
+            var deploymentConfig = $scope.attach.deploymentConfig;
+            var deployment = $scope.attach.deployment;
+            var podTemplate = deploymentConfig ? deploymentConfig.spec.template : deployment.spec.template;
+
+            var persistentVolumeClaim = $scope.attach.persistentVolumeClaim;
+            var name = $scope.attach.volumeName;
+            var mountPath = $scope.attach.mountPath;
+
+            // we must attach to at least one container (han-picked by user, or just "all")
+            if (!containerToAttachProvided()) {
+              sendError("You must specify at least one container.");
+              return;
+            }
+
+            // check the volume name wanted was not yet used in this pod template
+            if (!volumeNameIsUnique(name, podTemplate)) {
+              sendError("Volume named '" + name + "' already exists. Please choose another name.");
+              return;
+            }
+
+            if (mountPath) {
```

Comment:
we are requiring you to have a container selected to mount to... but not requiring you to have a mountPath set (and not defaulting otherwise)? that's a bit confusing.  Are we making mounting it in optional, or not.